golem-component-service-base/src/api/mapper.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::api::dto;
+use crate::model as domain;
+use crate::service::plugin::{PluginError, PluginService};
+use futures::{stream, StreamExt, TryStreamExt};
+use golem_common::model::component::ComponentOwner;
+use golem_common::model::plugin::{PluginInstallation, PluginScope};
+
+#[async_trait::async_trait]
+pub trait ApiMapper<Owner: ComponentOwner>: Send + Sync {
```

Comment:
This thing translates between our internal domain and the public api we want to expose to users.